boot/bootstate20.go

```diff
@@ -302,10 +355,10 @@ func (bs20 *bootState20Base) revisions() (curSnap, trySnap snap.PlaceInfo, tryin
 		return nil, nil, "", fmt.Errorf("cannot get snap revision: modeenv base boot variable is invalid: %v", err)
 	}
 
-	if bs20.modeenv.BaseStatus == TryingStatus && bs20.modeenv.TryBase != "" {
+	if bs20.modeenv.BaseStatus != DefaultStatus && bs20.modeenv.TryBase != "" {
```

Comment:
can you explain this change? does it affect old code?

Reply:
see the commit message for this change: https://github.com/snapcore/snapd/pull/8340/commits/3b5dfcb76fe3883262cefbaf22c0553d57b1b22b

specifically, during normal running status, snapd will only want the try base snap if we are in "trying", because it will already have transitioned to "trying" during the initramfs. However when we are in the initramfs and using this same code, we only want the try base snap when we are in "try", because we haven't transitioned it yet (indeed we are in the process of doing that). So, we can relax this check a bit to just check that it's not in default status, meaning it could be in try or trying and we return it and let the higher level function work out the logic by inspecting BaseStatus themselves (which they do in both cases)